Re-enabled the documentation module.

sbordet opened the pull request:
Now -DskipTests will compile the documentation classes but not build the documentation, saving time when building.

Much more work to do to update the documentation for Jetty 12.

Signed-off-by: Simone Bordet <simone.bordet@gmail.com>

sbordet committed:
Re-enabled the documentation module.

Now -DskipTests will compile the documentation classes but not build the documentation, saving time when building.

Much more work to do to update the documentation for Jetty 12.

Signed-off-by: Simone Bordet <simone.bordet@gmail.com>

olamy commented on documentation/jetty-documentation/pom.xml:
```diff
@@ -70,6 +70,7 @@
               <goal>process-asciidoc</goal>
             </goals>
             <configuration>
+              <skip>${skipTests}</skip>
```
to avoid repeating this, you should try add a single entry in the `<properties>` section such:
```
<properties>
  <asciidoctor.skip>${skipTests}</asciidoctor.skip>
</properties>
```

sbordet committed:
Now using ee10 rather than ee9.
Updated sources to reference ee10 implementation classes.
Fixed POM as suggested from review from @olamy.

Signed-off-by: Simone Bordet <simone.bordet@gmail.com>
